src/+replab/+str/Table.m

```diff
@@ -0,0 +1,365 @@
+classdef Table < replab.Str
+% Helper to display tables on the REPL
+%
+% >>> replab.str.Table(cell(4,4), 2:3, 'llcc') % unreadable
+
+    properties
+        rowSep % cell(1,nRows+1) of charstring: Row separators
+        colSep % cell(1,nCols+1) of charstring: Column separators
+               %
+               %                                This includes a separator on the left of the first column and
+               %                                on the right of the last column.
+        colAlign % char(1,nCols): Array of 'l' 'c' 'r' (default to 'c')
+        omitRange % integer(1,\*): Range of columns that can be omitted if space is lacking
+                  %
+                  %                This can be empty.
+        elements % cell(\*,\*): Table contents
+        columnLengths % integer(1, nCols): character width of each column in table
+        rowName % {1, 0}: whether table has row names
+        colName % {1, 0}: whether table has column names
+    end
+
+    
+    
+    methods
+
+        function self = Table(elements, varargin)
+            if ~iscell(elements) && ismatrix(elements)
+                elements = num2cell(elements);
+            elseif ~iscell(elements)
+                err('Error: elements must be matrix or cell array')
+            end
+            self.colName = 0;
+            self.rowName = 0;
+            s = struct(varargin{:});
+            if isfield(s, 'Cols')
+                self.addColumnNames(s.Cols)
+                self.colName = 1;
+            end
+            if isfield(s, 'Rows')
+                self.addRowNames(s.Rows)
+                self.rowName = 1;
+            end
+            dim = size(elements);
+            self.elements = self.convertToChars(elements);
```

Comment:
Here, I would group the `varargin` support, put it at the end of the constructor.